### ISSUE TITLE
md-list-item disabled property

### ISSUE THREAD

**Issue by theunreal** [angular/components#763]
Bug, feature request, or proposal:

Proposal

What is the expected behavior?

Angular material2 list **[disabled]** item. An option to bind `[disabled]` to `md-list-item` in order to disable it (same as how the button disabled property works).

What is the current behavior?

no [disabled] property binding with md-list-item

What is the use-case or motivation for changing an existing behavior?

A list is not a button, but used for navigation and accessing data. [disabled] property can be very useful in such cases.

Which versions of Angular, Material, OS, browsers are affected?

All

**Comment by jelbourn** [comment]
In general, `disabled` should live on the appropriate `<button>` / `<a>` elements, but it might make sense to have a style for disabled list items.

cc @kara

**Comment by code-tree** [comment]
This would be nice to have.

My use case: settings page, using md-list for settings, and some of which may be disabled.

**Comment by moneal** [comment]
This makes a lot of sense to me. I have a md-nav-list and being able to disable some options depending on account settings seems appropriate. I think this would function the same as md-menu.

**Comment by timmyrosen** [comment]
+1

**Comment by mrburrito** [comment]
+1

I'm using md-nav-list as a menu and need to be able to disable some or all of the list based on application state. I tried using md-button as the list items, but it doesn't render properly.

**Comment by Maccodonaldo** [comment]
+1

**Comment by gatimus** [comment]
I understand that the native anchor element has no `disabled` attribute but its hard to follow why `a[mat-button]` and `a[mat-menu-item]` use `disabled` input but `a[mat-list-item]` does not.

**Comment by cmd-johnson** [comment]
My workaround for now is to handle this through CSS:
```CSS
a[mat-list-item].disabled {
  opacity: 0.2;
  pointer-events: none;
}
```

```html
<mat-nav-list>
  <a mat-list-item>Clickable</a>
  <a mat-list-item class="disabled">Disabled</a>
</mat-nav-list>
```

It's not the same as a `disabled` attribute, but good enough for my use case.

**Comment by jelbourn** [comment]
Closing this since `mat-action-list` exists now which can have disable items

**Comment by william-lohan** [comment]
Even `mat-action-list` list doesn't seem to have `disabled` style

```html
<mat-action-list>
  <button mat-list-item (click)="save()" disabled> Save </button>
  <button mat-list-item (click)="undo()"> Undo </button>
</mat-action-list>
```

![image](https://user-images.githubusercontent.com/8146733/68042520-51d64d80-fc90-11e9-9ff6-0176a389fb83.png)

I'm using this

```ts
@Directive({
  selector: 'button[mat-list-item]',
  host: {
    '[class.mat-list-item-disabled]': 'disabled',
    '[attr.aria-disabled]': 'disabled',
    '[attr.disabled]': 'disabled || null'
  },
  inputs: ['disabled']
})
export class ListItemDisabledButtonStylerDirective extends _ListItemDisabledStylerDirectiveMixinBase implements CanDisable { }
```

to add selection list's `mat-list-item-disabled`

![image](https://user-images.githubusercontent.com/8146733/68042611-8fd37180-fc90-11e9-9508-8556b3027c4a.png)

and for `mat-nav-list`

``ts
@Directive({
  selector: 'mat-list-item, a[mat-list-item]',
  host: {
    '[class.mat-list-item-disabled]': 'disabled',
    '[attr.aria-disabled]': 'disabled'
  },
  inputs: ['disabled']
})
export class ListItemDisabledDirective extends _ListItemDisabledStylerDirectiveMixinBase implements CanDisable, OnChanges {

  constructor(
    @Optional() private matListItem?: MatListItem
  ) {
    super();
  }

  public ngOnChanges(changes: SimpleChanges): void {
    if (changes['disabled'] && this.matListItem) { // TODO matListItem null ?
      this.matListItem.disableRipple = this.disabled;
    }
  }

  @HostListener('click', ['$event'])
  public clickHandler(event: Event): boolean {
    if (this.disabled) {
      event.preventDefault();
      return false;
    }
    return true;
  }

}
```

https://stackblitz.com/edit/components-issue-th1faq-mat-list-item-disabled?file=app%2Flist-item-disabled-styler.directive.ts

**Comment by jelbourn** [comment]
@william-lohan good catch, I filed #17574
